src/action-creators/undoArchive.js

```diff
@@ -39,11 +36,12 @@ export const undoArchive = ({ originalPath, currPath, offset }) => dispatch => {
     offset
   })
 
-  if (getThoughtsRanked(getState(), archiveContext).length === 0) {
+  // Check if =archive is empty
+  if (getThoughtsRanked(getState(), context).length === 0) {
```

Comment:
When the order doesn't matter, use `getThoughts`, which is `O(1)`, instead of `getThoughtsRanked`.